src/etc/inc/filter.inc

```diff
@@ -1793,10 +1793,18 @@ function filter_nat_rules_generate_if ($if, $src = "any", $srcport = "", $dst =
 			$tgt = "{$natip}/32";
 		}
 	} else {
-		$natip = get_interface_ip($if);
-		if (is_ipaddr($natip)) {
-			$tgt = "{$natip}/32";
+		if (is_subnetv6($src) || is_subnetv6($dst)) {
+			$natip = get_interface_ipv6($if);
+			if (is_ipaddrv6($natip)) {
+				$tgt = "{$natip}/128";
+			}
 		} else {
```

Comment:
This should probably test for IPv4, in case it's a broken IPv6 scenario. Then add another `else` to deal with a case that can't be resolved properly, so that it never makes a broken rule.